### PR TITLE
[flutter_tools] remove even more mocks

### PR DESCRIPTION
**Pull request by jonahwilliams**
Work towards #71511

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/base/process_test.dart [hunk 1]
```diff
@@ -189,68 +161,6 @@ void main() {
         throwsA(isA<ProcessException>()),
       );
     });
-
-    testWithoutContext(' flaky process fails without retry', () async {
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/base/process_test.dart:193]: These tests have been skipped for months, lets delete and start again another time

File: packages/flutter_tools/test/general.shard/ios/xcodeproj_test.dart [hunk 2]
```diff
@@ -16,55 +16,17 @@ import 'package:flutter_tools/src/ios/xcode_build_settings.dart';
...
 
 import '../../src/common.dart';
 import '../../src/context.dart';
-import '../../src/mocks.dart' as mocks;
 
 const String xcodebuild = '/usr/bin/xcodebuild';
 
 void main() {
   group('MockProcessManager', () {
-    mocks.MockProcessManager processManager;
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/ios/xcodeproj_test.dart:29]: We shouldn't be doing the timeout and restart anymore

File: packages/flutter_tools/test/general.shard/resident_web_runner_cold_test.dart [hunk 3]
```diff
@@ -138,55 +178,63 @@ void main() {
       BuildResult(success: true),
       BuildResult(success: false),
     ]),
-    FileSystem: () => MemoryFileSystem.test(),
+    FileSystem: () => fileSystem,
     ProcessManager: () => FakeProcessManager.any(),
   });
+}
 
-  testUsingContext('Correctly performs a full refresh on attached chrome device.', () async {
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/resident_web_runner_cold_test.dart:145]: this is testing a mock

File: packages/flutter_tools/test/general.shard/base/process_test.dart [hunk 4]
```diff
@@ -4,21 +4,15 @@
...
 import 'package:flutter_tools/src/base/platform.dart';
 import 'package:flutter_tools/src/base/process.dart';
 import 'package:flutter_tools/src/base/terminal.dart';
-import 'package:mockito/mockito.dart';
 
 import '../../src/common.dart';
 import '../../src/fake_process_manager.dart';
 import '../../src/fakes.dart';
-import '../../src/mocks.dart' show MockProcessManager,
-                                   flakyProcessFactory;
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/base/process_test.dart:21]: `MockProcessManager`, `flakyProcessFactory`, and `_ProcessFactory` look dead on this PR, if you'd like the joy of deleting those as well.

In fact if you move `MockAndroidSdk` into `android_sdk_test` you can delete mocks.dart totally.

**Review by jmagman — APPROVED** [review]
LGTM, removing dead code from mocks.dart can be addressed in another PR, up to you.

File: packages/flutter_tools/test/general.shard/base/process_test.dart (hunk 4 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/base/process_test.dart:21]: Ahh thats right, I already landed the PR that removed that usage. Will update

**Pull request merged**